### DISCUSSION TITLE
diesel does not connection to postgres database with `sslmode` required

### DISCUSSION THREAD

**Discussion by ghost** [diesel-rs/diesel#3326] [Q&A]
### Versions

- **Rust:** stable 11 sept 2022
- **Diesel:** 2.0.0
- **Database:** postgres
- **Operating System** mac m1

### Feature Flags

- **diesel:**
- `["postgres","numeric","chrono"]`

## Problem Description
diesel-cli runs migrations successfully with `sslmode=require` attached to database url
but when performing `diesel_insert` in program, it says `error performing TLS handshake: no TLS implementation configured"`

### What are you trying to accomplish?
Connect to postgres database with `sslmode` required. 

### What is the expected output?
Successful database insert

### What is the actual output?
`error performing TLS handshake: no TLS implementation configured"`

### Are you seeing any additional errors?
N/A

### Steps to reproduce
Setup postgres 14 database with encryption only connection.
Create insert query using diesel with `sslmode` required

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues)  and the [disussion forum](https://github.com/diesel-rs/diesel/discussions)  for similar possible closed issues.
 
https://github.com/sgrif/pq-sys/issues/25
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [x] This issue can be reproduced without requiring a third party crate

**Comment by ghost** [comment]
Also adding `allow` to `sslmode` url query results in

`UnableToSendCommand, "invalid connection string: invalid value for option sslmode`

**Comment by weiznich** [comment]
This sounds like an issue with your environment. Can you verify that `psql postgres://your/db_url` works for you. If not that indicates that your database url does just not match your setup.

**Reply by weiznich** [reply]
(Converted to a discussion as this seems more like a support case for me)

**Comment by ghost** [comment]
Turns out that diesel_async does not support postgres tls - https://github.com/weiznich/diesel_async/blob/main/src/pg/mod.rs#L124

**Reply by weiznich** [reply]
Just to clarify a few things: 

1. `diesel_async` is not `diesel`. Please do not report bugs for the first crate here in this repository, as it is currently not part of the official set of diesel crates. Such reports are considered off topic.
2. There is not even a first alpha release of `diesel_async` yet so please expect bugs and missing features. Additionally we do not provide any support for unreleased crates, so your own your own there (although we appropriate well written reproducible bug reports, so be sure to include all relevant information in potential reports)
3. It's just not true that `diesel_async` does not support postgres tls, you just manually need to construct a connection via different methods than `establish`.

**Reply by ghost** [reply]
Noted 👍 and kind of what I did here https://github.com/lilses/diesel_async/commit/0a45f631913a078abd377bdabf6a7236ac58d48f. Would submit a pull request - would that require a new trait other than `AsyncConnectionGatWorkaround` to add new fn `establish_tls` for example? It would be for PG only, MySQL works with ssl fine.

**Reply by ghost** [reply]
@weiznich can you reply or close this discussion

**Reply by weiznich** [reply]
> diesel_async is not diesel. Please do not report bugs for the first crate here in this repository, as it is currently not part of the official set of diesel crates. Such reports are considered off topic.

and 

> It's just not true that diesel_async does not support postgres tls, you just manually need to construct a connection via different methods than establish.

To explicitly link the relevant method: [`AsyncPgConnection::try_from`](https://docs.rs/diesel-async/0.1.0/diesel_async/struct.AsyncPgConnection.html#method.try_from)
